FeedAPIChallenge/Feed API/RemoteFeedLoader.swift

```diff
@@ -19,6 +19,47 @@ public final class RemoteFeedLoader: FeedLoader {
 	}
 
 	public func load(completion: @escaping (FeedLoader.Result) -> Void) {
-		fatalError("Must be implemented")
+		client.get(from: url) { [weak self] result in
+			guard self != nil else { return }
+			completion(Swift.Result { try Self.mapClientResult(result) })
+		}
+	}
+
+	private static func mapClientResult(_ clientResult: HTTPClient.Result) throws -> [FeedImage] {
+		switch clientResult {
+		case .failure:
+			throw Error.connectivity
+		case .success((let data, let httpResponse)):
+			guard httpResponse.statusCode == 200 else {
+				throw Error.invalidData
+			}
+			do {
+				let remoteItems = try JSONDecoder().decode(Items.self, from: data)
+				return remoteItems.items.map { $0.makeEquivalentFeedImage() }
+			} catch {
+				throw Error.invalidData
+			}
```

Comment:
Since the error that is thrown by `JSONDecoder().decode` is ignored and simply replaced with `Error.invalidData`, `do { try JSONDecoder().decode } ... catch` block can be replaced with optional try like this:
```
guard let remoteItems = try? JSONDecoder().decode(Items.self, from: data) else {
    throw Error.invalidData
}
```

And if so, then two `guard ...` blocks can be combined into one to make the code shorter

```suggestion
			guard httpResponse.statusCode == 200, let remoteItems = try? JSONDecoder().decode(Items.self, from: data) else {
				throw Error.invalidData
			}
			return remoteItems.items.map { $0.makeEquivalentFeedImage() }
```